### PR TITLE
Allow seed deletion in combined garden runtime cluster and soil cluster setup

### PR DESCRIPTION
**Pull request by ScheererJ**
**How to categorize this PR?**

/area monitoring
/area networking
/kind bug

**What this PR does / why we need it**:

Allow seed deletion in combined garden runtime cluster and soil cluster setup.

Two managed resources included the same resource. One of them was managed by gardener operator, the other one was in gardenlet's responsibility. If gardener operator was deployed to a cluster that later became a seed, deletion of the seed was ending in a race between both gardener resource managers fighting over the single resource (deleting/recreating them in the process).
As the resource is not useful at all in the garden runtime cluster, it will only be deployed into seed clusters going forward.

**Which issue(s) this PR fixes**:
None.

**Special notes for your reviewer**:

**Release note**:

```other operator
It is now possible to delete the seed in a combined garden runtime/soil setup.
```

### CODE REVIEW

**Review by LucaBernstein — COMMENTED** [review]
/lgtm

File: pkg/component/networking/istio/istiod.go [hunk 1]
```diff
@@ -251,54 +251,57 @@ func (i *istiod) Deploy(ctx context.Context) error {
...
-			Endpoints: []monitoringv1.Endpoint{{
-				Path: "/stats/prometheus",
-				Port: "tls-tunnel",
-				RelabelConfigs: []monitoringv1.RelabelConfig{{
-					SourceLabels: []monitoringv1.LabelName{"__meta_kubernetes_pod_ip"},
-					Action:       "replace",
-					TargetLabel:  "__address__",
-					Regex:        `(.+)`,
-					Replacement:  ptr.To("${1}:15020"),
+	// Only add a ServiceMonitor for the istio-ingressgateway if istio is deployed on a seed.
```
Comment (rfranzke) [inline pkg/component/networking/istio/istiod.go]: Can you add an explanation why this is done?

Comment (ScheererJ) [reply pkg/component/networking/istio/istiod.go]: Addressed in https://github.com/gardener/gardener/pull/10143/commits/7eab172661b97efe57189763e38ea1e42617e667. PTAL whether this is what you wanted.

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**